.github/workflows/auto-translate.yml

```
@@ -0,0 +1,94 @@
+name: Auto Translate
+
+on:
+  workflow_call:
+    inputs:
+      text_domain:
+        description: 'Text domain for translation'
+        required: true
+        type: string
+
+    secrets:
+      TRANSLATOR_API_KEY:
+        required: true
+      NEWFOLD_ACCESS_TOKEN:
+        required: true
+
+permissions: {}
+
+jobs:
+  auto-translate:
+    name: Auto-Translate and Create PR
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      pull-requests: write
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+        with:
+          persist-credentials: false
+
+      - name: 'Check for existing translations PR and exit if found'
+        id: check_pr
+        env:
+          GH_TOKEN: ${{ secrets.NEWFOLD_ACCESS_TOKEN }}
+        run: |
+          prs=$(gh pr list --state open --label 'translations' --json number --jq '.[].number')
+          if [ -n "$prs" ]; then
+            echo 'A PR with "translations" label already exists:'
+            echo "$prs"
+            echo "found=true" >> $GITHUB_OUTPUT
+          else
+            echo 'No existing "translations" PR found. Continuing workflow.'
+            echo "found=false" >> $GITHUB_OUTPUT
+          fi
+
+      - name: Set up Python
+        if: steps.check_pr.outputs.found == 'false'
+        uses: actions/setup-python@a26af69be951a213d495a4c3e4e4022e16d87065 # v5.6.0
+        with:
+          python-version: '3.x'
+
+      - name: Install Python dependencies
+        if: steps.check_pr.outputs.found == 'false'
+        run: |
+          pip install requests polib
+
+      - name: Set branch name
+        if: steps.check_pr.outputs.found == 'false'
+        id: vars
+        run: echo "BRANCH=auto/translated-$(date +%s)" >> "$GITHUB_OUTPUT"
+
+      - name: Download translate.py from parent repo
+        if: steps.check_pr.outputs.found == 'false'
+        run: |
+          mkdir -p .github/scripts
+          curl -H "Authorization: token ${{ secrets.NEWFOLD_ACCESS_TOKEN }}" \
+              -H "Accept: application/vnd.github.v3.raw" \
+              -o .github/scripts/translate.py \
+              https://api.github.com/repos/newfold-labs/workflows/contents/.github/scripts/translate.py?ref=main
```

This works, but I think we can avoid the need for cURL or an access token.

The `sparse-checkout` argument for `actions/checkout` allows you to only checkout specific files or directories based on pattern matching.

```suggestion
      - name: Download translate.py from parent repo
        if: steps.check_pr.outputs.found == 'false'
        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
        with:
              repository: 'newfold-labs/workflows'
              path: 'translation-action-scripts'
              sparse-checkout: '.github/scripts/translate.py'
              sparse-checkout-cone-mode: false
              persist-credentials: false
```
We should also avoid using a folder named `.github` in the root directory just to avoid any possible problems overwriting files present in a calling repository. `translation-action-scripts` should be relatively safe.